build/ical.js

```diff
@@ -6974,12 +6974,26 @@ ICAL.RecurIterator = (function() {
         throw new Error("BYYEARDAY may only appear in YEARLY rules");
       }
 
+
+      this.sort_bypart_rules("BYSECOND", parts);
       this.last.second = this.setup_defaults("BYSECOND", "SECONDLY", this.dtstart.second);
+
+      this.sort_bypart_rules("BYMINUTE", parts);
       this.last.minute = this.setup_defaults("BYMINUTE", "MINUTELY", this.dtstart.minute);
+
+      this.sort_bypart_rules("BYHOUR", parts);
       this.last.hour = this.setup_defaults("BYHOUR", "HOURLY", this.dtstart.hour);
-      this.last.day = this.setup_defaults("BYMONTHDAY", "DAILY", this.dtstart.day);
+
+      this.sort_bypart_rules("BYMONTH", parts);
       this.last.month = this.setup_defaults("BYMONTH", "MONTHLY", this.dtstart.month);
 
+      //setting day after month, as sorting BYMONTHDAY will differ based on the month
```

Comment:
If sorting bymonthday differs based on month, wouldn't we need to sort this differently every month while the rule is iteraed?